registrations/tasks.py

```diff
@@ -375,6 +377,79 @@ def run(self, registration_id, **kwargs):
 validate_subscribe = ValidateSubscribe()
 
 
+class PushRegistrationToJembi(Task):
+    """ Task to push registration data to Jembi
+    """
+    name = "ndoh_hub.registrations.tasks.push_registration_to_jembi"
+    l = get_task_logger(__name__)
+
+    def get_patient_id(self, id_type, id_no=None,
+                       passport_origin=None, mom_msisdn=None):
+        if id_type == 'sa_id':
+            return id_no + "^^^ZAF^NI"
+        elif id_type == 'passport':
+            return id_no + '^^^' + passport_origin.upper() + '^PPN'
+        else:
+            return mom_msisdn.replace('+', '') + '^^^ZAF^TEL'
+
+    def get_subscription_type(self, authority):
+        authority_map = {
+            'personal': 1,
+            'chw': 2,
+            'clinic': 3
+        }
+        return authority_map[authority]
+
+    def get_today(self):
+        return datetime.today()
+
+    def get_timestamp(self):
+        return self.get_today().strftime("%Y%m%d%H%M%S")
+
+    def get_dob(self, mom_dob):
+        if mom_dob is not None:
+            return mom_dob.strftime("%Y%m%d")
+        else:
+            return None
+
+    def build_jembi_json(self, registration):
+        """ Compile json to be sent to Jembi. """
+        "TODO: confirm these values are mapped correctly from ndoh-control"
+        json_template = {
+            "mha": 1,
+            "swt": 1,
+            "dmsisdn": registration.hcw_msisdn,
+            "cmsisdn": registration.mom_msisdn,
+            "id": self.get_patient_id(
+                registration.mom_id_type, registration.mom_id_no,
+                registration.mom_passport_origin, registration.mom_msisdn),
+            "type": self.get_subscription_type(registration.authority),
+            "lang": registration.mom_lang,
+            "encdate": self.get_timestamp(),
+            "faccode": registration.clinic_code,
+            "dob": self.get_dob(registration.mom_dob)
+        }
+
+        # Self registrations on all lines should use cmsisdn as dmsisdn too
+        if registration.hcw_msisdn is None:
+            json_template["dmsisdn"] = registration.mom_msisdn
+
+        if registration.authority == 'clinic':
+            json_template["edd"] = registration.mom_edd.strftime("%Y%m%d")
+
+        return json_template
```

Comment:
@jpaidoussi could I get your input on this? This is obviously failing because I'm trying to repopulate a JSON payload for an API request using a new style registration and the fields either don't exist or aren't of the same name. Could you help me with how these should be mapped?